### PR TITLE
Use NativeLibrary for all OSes

### PR DESCRIPTION
**Pull request by imnasnainaec**

### CODE REVIEW

File: source/icu.net/NativeMethods/NativeMethods.cs [hunk 1]
```diff
@@ -90,16 +90,6 @@ static NativeMethods()
 		[DllImport(LIBDL_NAME, SetLastError = true)]
 		private static extern IntPtr dlsym(IntPtr handle, string name);
 
-		[DllImport(LIBDL_NAME, EntryPoint = "dlerror")]
-		private static extern IntPtr _dlerror();
-
-		private static string dlerror()
```
Comment (lyonsil) [inline source/icu.net/NativeMethods/NativeMethods.cs:96]: This was used when logging load errors on Linux. It's not important in the successful cases, but I don't know how helpful it might be when debugging for some people/situations. Why did you remove it?

**Review by lyonsil — COMMENTED** [review]
At a high level these changes make sense to me. Other than the one thing I pointed out with `dlerror`, my only concern is how much testing this might need on Windows and Linux to make sure there isn't some unexpected impact. `NativeLibrary` should work fine, but I don't think the unit tests really exercise dynamic library loading deeply (e.g., does `NativeLibrary` follow all the same path searching rules that `libdl` uses?). I'm guessing it will not be a problem, but that's just a guess.

File: source/icu.net/NativeMethods/NativeMethods.cs (hunk 1 above)
Comment (imnasnainaec) [reply source/icu.net/NativeMethods/NativeMethods.cs:96]: Since it was no longer relevant in the `#if NET6_0_OR_GREATER` situation, and for the sake of having a common error line for all the OSes:
```
				lastError = Marshal.GetLastWin32Error();
				Trace.WriteLineIf(lastError != 0, $"Unable to load [{libPath}]. Error: {lastError}");
				Trace.TraceWarning($"{loadMethod} of {libPath} failed with error {lastError}");
```

Comment (imnasnainaec) [reply source/icu.net/NativeMethods/NativeMethods.cs]: OS-specific error messages restored.

**Review by lyonsil — APPROVED** [review]
Looks good to me. The overall question about "how do we know that NativeLibrary will be functionally compatible with the previous approach for Windows and Linux users on .NET 6.0 or later?" is still there, but I don't see anything in these changes that looks like a regression or a problem.

**Comment by imnasnainaec** [comment]
> Looks good to me. The overall question about "how do we know that NativeLibrary will be functionally compatible with the previous approach for Windows and Linux users on .NET 6.0 or later?" is still there, but I don't see anything in these changes that looks like a regression or a problem.

I'll try to test it out more.

For reference, here's the code `NativeLibrary` uses: [for Unix](https://github.com/dotnet/runtime/blob/5c53e8ec9ca1a06a31c45936d3058fd6b1dc5961/src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.NativeAot.Unix.cs#L12); [for Windows](https://github.com/dotnet/runtime/blob/5c53e8ec9ca1a06a31c45936d3058fd6b1dc5961/src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.NativeAot.Windows.cs#L12)

**Comment by imnasnainaec** [comment]
For Unix, it uses `[LibraryImport(Interop.Libraries.SystemNative, EntryPoint = "SystemNative_LoadLibrary", StringMarshalling = StringMarshalling.Utf8)]` (in [Interop.Sys](https://github.com/dotnet/runtime/blob/f88ba03002108ee47fc0cafce8e91b06032278e9/src/libraries/Common/src/Interop/Unix/System.Native/Interop.DynamicLoad.cs#L11))

**Review by jasonleenaylor — COMMENTED** [review]
I verified (using a locally built package with FieldWorks) that this code does not disrupt the framework clients of the .dll, and I don't see any reason that it shouldn't work for dotnet clients on Windows.
